app/api_1_0/url.py

```diff
@@ -0,0 +1,388 @@
+def shorturls():
+    """
+    This function returns a collection of shortened URLs by the current user
+    """
+    user_urls = ShortUrl.query.filter_by(
+        user=g.current_user).filter_by(deleted=0).all()
+    if user_urls:
+        urls = []
+        for i in range(len(user_urls)):
+            urls.append(user_urls[i].get_details())
+        return jsonify({'success': True, 'urls': urls})
+    return not_found('No shortened url found')
+
+
+@api.route('/shorturl/<int:id>', methods=['GET', 'PUT', 'DELETE'])
+def shorturl(id):
+    """
+    This function handles deleting, updating and getting the details of a
+    short URL by mapping request method to the appropraite function.
+
+    Keyword arguments:
+    id -- the integer value for the unique identification of the URL
+    """
+    request_mapping = {'GET': get_short_url_details,
+                       'DELETE': delete_short_url,
+                       'PUT': change_short_url_target}
+    url = ShortUrl.query.filter_by(user=g.current_user).filter_by(
+        short_url_id=id).filter_by(deleted=0).first()
+    if not url:
+        return not_found(
+            "No shortened url with id '{}' found for you".format(id))
+    return request_mapping[request.method](url)
+
+
+def get_short_url_details(url):
+    """
+    This function returns the details of a shorl URL
+
+    Keyword arguments:
+    url -- the short url object for which details will be returned
+    """
+    return jsonify({'success': True, 'url': url.get_details()})
+
+
+def delete_short_url(url):
+    """
+    This function deletes a shorl URL by setting its deleted propert to 1.
+
+    Keyword arguments:
+    url -- the short url object to be deleted
+    """
+    count = len(ShortUrl.query.filter_by(long_url=url.long_url).all())
+    if count <= 1:
+        LongUrl.query.filter_by(long_url=url.long_url.long_url).delete()
+    url.deleted = 1
+    db.session.commit()
+    return jsonify({'success': True, 'message': 'deleted'})
+
+
+def change_short_url_target(url):
+    """
+    This function updates the long URL of a short URL with the long URL
+    supplied in the body of the request.
+
+    Keyword arguments:
+    url -- the short url object to be updated
+    long_url -- the string value to be updated in the short URL
+    """
+    long_url_input = ValidateLongUrl(request)
+    if not long_url_input.validate():
+        return bad_request(long_url_input.errors)
+    new_long_url = request.json.get('long_url')
+    has_short_url = ShortUrl.query.filter_by(
+        user=g.current_user).filter(ShortUrl.long_url.has(
+            long_url=new_long_url)).filter_by(deleted=0).first()
+    if has_short_url:
+        return forbidden("The url already has a shortened url '{}'".format(
+            has_short_url.short_url))
+    count = len(ShortUrl.query.filter_by(
+        long_url=url.long_url).filter_by(deleted=0).all())
+    if count <= 1:
+        LongUrl.query.filter_by(long_url=url.long_url.long_url).delete()
```

Comment:
The url.long_url is an object that has an attribute long_url and other attributes such as long_url_id and short_urls.